src/main/java/com/google/firebase/messaging/AndroidNotification.java

```diff
@@ -309,6 +320,24 @@ public Builder setImage(String imageUrl) {
       return this;
     }
 
+    /**
+     * Sets the number of items this notification represents. 
+     * May be displayed as a badge count for launchers that support badging.
+     *
+     * <p>If not invoked then notification count is left unchanged
+     * 
+     * @see <a 
+     * href="https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#androidnotification">
+     * Android Notification Spec</a>
+     * 
+     * @param notificationCount 0=reset, positive=count
```

Comment:
Can you spell this out instead of this shorthand form? "Zero or positive integer. Zero indicates leave unchanged."

Reply:
Ok but ZERO does not indicate leave unchanged, zero means reset or clear the notification badge on the users device. This is ultimately why we were having the other conversation about setNotificationCount(null), the idea was that it would not render out the notification_count parameter in the resultant json and so would be interpreted as unchanged.